drivers/linux/chipsec_km.c

```diff
@@ -1676,6 +1681,94 @@ static struct miscdevice chipsec_dev = {
     .fops = &mem_fops
 };
 
+/*
+ * 0ld dog never die:
+ * https://gist.githubusercontent.com/GoldenOak/a8cd563d671af04a3d387d198aa3ecf8/raw/8dcc90dbbf9b9ffd65cc2c03f1cd48445b84c2b6/obtain_syscall_table_by_proc.c
+*/
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0)
+
+unsigned long chipsec_lookup_name(const char *name)
+{
+	char *file_name                       = KALLSYMS_PATH;
+	int i                                 = 0;         /* Read Index */
+	struct file *proc_ksyms               = NULL;      /* struct file the '/proc/kallsyms' or '/proc/ksyms' */
+	char *sct_addr_str                    = NULL;      /* buffer for save sct addr as str */
+	char proc_ksyms_entry[BUFF_SIZE]  = {0};       /* buffer for each line at file */
+	unsigned long* res                    = NULL;      /* return value */
+	char *proc_ksyms_entry_ptr            = NULL;
+	int read                              = 0;
+	int err = 0;
+	mm_segment_t oldfs;
+
+
+	/* Allocate place for sct addr as str */
+	if((sct_addr_str = (char*)kmalloc(BUFF_SIZE * sizeof(char), GFP_KERNEL)) == NULL)
+		goto CLEAN_UP;
+
+	proc_ksyms = filp_open(file_name, O_RDONLY, 0);
+       	if (proc_ksyms == NULL)
+		goto CLEAN_UP;
+
+	oldfs = get_fs();
+	set_fs (KERNEL_DS);
+	read = proc_ksyms->f_op->read(proc_ksyms, proc_ksyms_entry + i, 1, &(proc_ksyms->f_pos));
+	set_fs(oldfs);
+
+	while( read == 1)
+	{
+		if(proc_ksyms_entry[i] == '\n' || i == BUFF_SIZE)
+		{
+			if(strstr(proc_ksyms_entry, name) != NULL)
+			{
+
+                        	printk(KERN_INFO"[+] %s: %s", name, proc_ksyms_entry);
+
+				proc_ksyms_entry_ptr = proc_ksyms_entry;
+				strncpy(sct_addr_str, strsep(&proc_ksyms_entry_ptr, " "), BUFF_SIZE);
+				if((res = kmalloc(sizeof(unsigned long), GFP_KERNEL)) == NULL)
+					goto CLEAN_UP;
+				err = kstrtoul(sct_addr_str, 16, res);
+				goto CLEAN_UP;
+			}
+
+			i = -1;
+			memset(proc_ksyms_entry, 0, BUFF_SIZE);
+		}
+
+		i++;
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
```

Comment:
Is this condition ever happening, given the LINUX_VERSION_CODE here should be higher than Kerner version 5.4.0 already?

Reply:
Those code should work on >= vanilla kernel v5.0.0 but it's clearly won't work under v5.4.0 since the commit is try to compatible w/ wider version of grsec. So what do you think if we change LINUX_VERSION_CODE to > v5.4.0( make sense to the current chipsec context) or just remain the original code there? I'm ok with both. Let me know your thought.

Reply:
I believe the issue is this "if" is inside the "if" on line 1688. If it's the kernel version is less than 5.4.0, it will use kallsyms_lookup_name, and skip this "if" all together.

Reply:
Removed the conditional compilation due to its no use in chipsec. Please review.

Reply:
Awesome. Thank you! I merged it in and it will be included in the next release.